### PR TITLE
Added gamepad maping

### PR DESCRIPTION
**Pull request by GamaDu30**

### CODE REVIEW

File: src/utils/gamepad.ts [hunk 1]
```diff
@@ -761,40 +770,80 @@ export class gamepads {
     this.delAllListeners(this._btnsListeners);
   }
 
+  removeInputCallbacks(inputName: string) {
```
Comment (Inateno) [inline src/utils/gamepad.ts:773]: where this should be used and how? It looks like it doesn't remove the listeners but just remove from the Map.

File: src/utils/gamepad.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
-﻿import EventEmitter from 'eventemitter3';
+﻿import Save from '@dreamirl/dreamengine/src/utils/Save';
+import EventEmitter from 'eventemitter3';
 import config from '../config';
 import Events from './Events';
-import InputsManager, { InputType, Inputs } from './Inputs';
+import Inputs, { InputType, default as InputsManager } from './Inputs';
```
Comment (Inateno) [inline src/utils/gamepad.ts:5]: why this ?

**Review by Inateno — CHANGES_REQUESTED** [review]

File: src/utils/gamepad.ts [hunk 3]
```diff
@@ -1,7 +1,8 @@
-﻿import EventEmitter from 'eventemitter3';
+﻿import Save from '@dreamirl/dreamengine/src/utils/Save';
+import EventEmitter from 'eventemitter3';
 import config from '../config';
 import Events from './Events';
-import InputsManager, { InputType, Inputs } from './Inputs';
+import Inputs, { InputType } from './Inputs';
```
Comment (liro-u) [inline src/utils/gamepad.ts:5]: Need to keep previous naming to prevent compatibility with old project for update

**Review by liro-u — CHANGES_REQUESTED** [review]

File: src/utils/gamepad.ts [hunk 4]
```diff
@@ -436,10 +442,10 @@ export class gamepads {
               continue;
             }
 
-            let key: keyof typeof InputsManager.dbInputs.GAMEPADBUTTONS;
+            let key: keyof typeof this.inputs.dbInputs.GAMEPADBUTTONS;
             let keyName: string | undefined = undefined;
-            for (key in InputsManager.dbInputs.GAMEPADBUTTONS) {
-              if (InputsManager.dbInputs.GAMEPADBUTTONS[key] == i) {
+            for (key in this.inputs.dbInputs.GAMEPADBUTTONS) {
+              if (this.inputs.dbInputs.GAMEPADBUTTONS[key] == i) {
```
Comment (liro-u) [inline src/utils/gamepad.ts:448]: use inputsmanager

**Review by liro-u — CHANGES_REQUESTED** [review]

**Review by liro-u — APPROVED** [review]

**Pull request merged**